Library/Homebrew/cmd/commands.rb

```diff
@@ -31,14 +31,18 @@ def commands
       return
     end
 
-    ohai "Built-in commands", Formatter.columns(Commands.internal_commands)
-    puts
-    ohai "Built-in developer commands", Formatter.columns(Commands.internal_developer_commands)
-
-    external_commands = Commands.external_commands
-    return if external_commands.blank?
-
-    puts
-    ohai "External commands", Formatter.columns(external_commands)
+    [["Built-in commands", -> { Commands.internal_commands }],
+     ["Built-in developer commands", -> { Commands.internal_developer_commands }],
+     ["External commands", -> { Commands.external_commands }],
+     ["Cask commands", -> { Commands.cask_internal_commands }],
+     ["External cask commands", -> { Commands.cask_external_commands }]]
+      .each_with_index do |title_and_proc, index|
+      title, proc = title_and_proc
+      cmds = proc.call
+      if cmds.present?
+        puts unless index.zero?
```

Comment:
How about using a Boolean variable and `||=` instead of needing the index? That could also let you iterate over a hash instead with `do |title, proc|`. I'm not sure I quite understand why to use a Proc here though rather than having the values evaluated immediately?